File: ruby/egoing's python and ruby/ex17_oop_classmember.py

class C:
    @staticmethod
    def stt_method():
        print('called static_method in C')

    @classmethod
    def cls_method(cls):
        print('called class_method in C')

    def ins_method(self):
        print('called instance_method in C')        


ins = C()
C.stt_method()
C.cls_method()
ins.ins_method()


ins.cls_method()

File: ruby/egoing's python and ruby/ex17_oop_classmember.rb

class C
  # class method
  def C.cls_method()
    puts 'called cls_method in C'
  end

  # instance method
  def ins_method()
    puts 'called ins_method in C'
  end
end

ins = C.new()
C.cls_method()
ins.ins_method()
